Enhance VP verification response schema.

bparth24 opened the pull request:
- Adds summary counts providing quick overview of credential verification results
- Implements detailed per-VC breakdown showing verification status for each credential
- Includes granular verification details for status checks, validity periods, proof verification, and schema validation
- Provides transparency about which VCs were checked vs skipped with explanatory reasons
- Adds comprehensive error reporting with problemDetails at presentation, credential, and proof levels
- Enables understanding of which proofs were verified and verification factors checked per credential
- Changes to file - `/components/VerifyPresentationResult.yml`

Addresses #509.

----
----

<img width="830" height="882" alt="Screenshot 2025-08-05 at 2 54 06 PM" src="https://github.com/user-attachments/assets/32e2656e-0b1a-4ecf-a49f-a6907f76b6d3" />

<img width="818" height="710" alt="Screenshot 2025-08-05 at 2 54 25 PM" src="https://github.com/user-attachments/assets/5bd7a2ea-3062-4805-9728-29af465b48b8" />

<img width="820" height="769" alt="Screenshot 2025-08-05 at 2 54 40 PM" src="https://github.com/user-attachments/assets/3dbc754c-55de-4554-9b91-ccbf60eaa868" />

<img width="814" height="658" alt="Screenshot 2025-08-05 at 2 54 50 PM" src="https://github.com/user-attachments/assets/54f07927-f15a-4cd2-a9ca-aa736ccf05f3" />

----
----

@dlongley @msporny @TallTed @PatStLouis - Ready for review

TallTed commented on components/VerifyPresentationResult.yml:
```diff
@@ -19,72 +19,303 @@ components:
         verified:
           type: boolean
-          description: Overall verification assertion of the VerifiablePresentation. This is set to `True` if no errors were detected during the verification process; otherwise, `False`. See error handling section for more guidance around errors, warnings, and validation and verification steps.
+          description: Overall verification assertion of the VerifiablePresentation. This is set to `True` if the VerifiablePresentation and all contained VerifiableCredentials passed verification; otherwise, `False`. See error handling section for more guidance around errors and warnings, and validation and verification steps.
```
"all contained" is only appropriate if all VCs within VP will be checked, i.e., if the user cannot select *some* VCs to check.
```suggestion
          description: Overall verification assertion of the VerifiablePresentation. This is set to `True` if the VerifiablePresentation and all checked VerifiableCredentials passed verification; otherwise, `False`. See error handling section for more guidance around errors and warnings, and validation and verification steps.
```

TallTed reviewed (changes requested):
Just a tiny reversion.

msporny commented on components/VerifyPresentationResult.yml:
```diff
@@ -19,72 +19,303 @@ components:
+                      verified:
+                        type: boolean
+                        description: Whether this specific VerifiableCredential pass verification. Only meaningful if wasChecked is true.
+                      skipReason:
```
Why would you skip checking a proof that is vital for security?

msporny commented on components/VerifyPresentationResult.yml:
```diff
@@ -19,72 +19,303 @@ components:
+                          type: object
+                          description: Results of validating a credentialStatus object.
+                          properties:
+                            value:
```
Hmm, this seems like you would need to pass in the expected status in the check... I don't think we can report on 'valid' or not without that. We might need something like: `expectedValue`.

msporny commented on components/VerifyPresentationResult.yml:
```diff
@@ -19,72 +19,303 @@ components:
-        }],
+        "problemDetails": [
+          {
+            "title": "PARSING_ERROR",
```
I don't think this is what we meant by "human-readable" string in the spec. Something like this might be better:

```json
{
  "type": "https://www.w3.org/TR/vc-data-model#PARSING_ERROR",
  "title": "Parsing error",
  "description": "There was a parsing error on line 32, column 5: A comma was expected after the \" character."
}
```

bparth24 replied:
@msporny 

> I don't think this is what we meant by "human-readable" string in the spec. Something like this might be better:
> 
> ```json
> {
>   "type": "https://www.w3.org/TR/vc-data-model#PARSING_ERROR",
>   "title": "Parsing error",
>   "description": "There was a parsing error on line 32, column 5: A comma was expected after the \" character."
> }
> ```

Addressed in commit - https://github.com/w3c-ccg/vc-api/commit/f4b15badd83dba5e5a7e642c1a7813168659178c

msporny commented on components/VerifyPresentationResult.yml:
```diff
@@ -19,72 +19,303 @@ components:
+            },
+            "proof": [{
+              "valid": true,
+              "input": {},
```
Why is this blank and not the proof that was checked?

msporny commented on components/VerifyPresentationResult.yml:
```diff
@@ -19,72 +19,303 @@ components:
+                    "input": {},
+                    "problemDetails": [
+                      {
+                        "title": "CRYPTOGRAPHIC_SECURITY_ERROR",
```
Not easily human readable

bparth24 replied:
@msporny 

> Not easily human readable

Addressed in commit - https://github.com/w3c-ccg/vc-api/commit/f4b15badd83dba5e5a7e642c1a7813168659178c

msporny commented on components/VerifyPresentationResult.yml:
```diff
@@ -19,72 +19,303 @@ components:
+                ],
+                "problemDetails": [
+                  {
+                    "title": "CREDENTIAL_EXPIRED",
```
Not easily human readable

bparth24 replied:
@msporny 

> Not easily human readable. See this for a better example:
> 
> https://www.rfc-editor.org/rfc/rfc9457#appendix-C
> 
> Remember that people will copy/paste these examples into production code. Any bad practice we provide here will be multiplied a hundred times over in deployed software.

Addressed in commit - https://github.com/w3c-ccg/vc-api/commit/f4b15badd83dba5e5a7e642c1a7813168659178c

msporny reviewed (changes requested):
Thank you for a PR that we can discuss, @bparth24.

There are significant design issues with the response format that we'll have to discuss during the next call. 

I also note inconsistencies in `True` vs `true`, use of backticks (which won't work in OAS files when rendered in ReSpec, you have to use &lt;code> tags instead, not referencing specs using HTTP URLs (VCDM v2), and not wrapping type names in code blocks. Please make a pass to fix all of that because it'll create spec debt that we'll just have to clean up later.

bparth24 commented on components/VerifyPresentationResult.yml:
```diff
@@ -19,72 +19,303 @@ components:
-                    type: boolean
-                    description: Results of validating the proof object.
-                  input:
+                summary:
+                  type: object
+                  description: Summary counts providing a quick overview of VerifiableCredential verification results.
+                  additionalProperties: false
+                  properties:
+                    totalCount:
+                      type: integer
+                      description: Total number of VerifiableCredentials in the VerifiablePresentation.
+                    checkedCount:
+                      type: integer
+                      description: Number of VerifiableCredentials that were actually verified.
+                    skippedCount:
+                      type: integer
+                      description: Number of VerifiableCredentials that were skipped due to options or configuration.
+                    passedCount:
+                      type: integer
+                      description: Number of checked VerifiableCredentials that passed verification.
+                    failedCount:
+                      type: integer
+                      description: Number of checked VerifiableCredentials that failed verification.
```
@msporny 

**On the summary redundancy:** I completely agree with your assessment. My initial instinct was the same - that the summary contains entirely derivable information that clients can calculate from the detailed results array. I don't have a strong preference for keeping it, and removing it would simplify the schema significantly.

**Regarding the skipping functionality:** I want to clarify that this wasn't intended for bypassing security checks. The skipping capability was included specifically because it was mentioned in Issue #284 as beneficial for troubleshooting and debugging scenarios where developers need to test only VP-level verification while isolating credential-specific issues.

However, I acknowledge your security concern. Depending on the workflow or use case, skipping proof checks could introduce complexity and potential misuse. 

**Next steps:** Remove the summary object and reconsider whether skipping functionality is necessary for the initial implementation.

bparth24 commented:
@msporny 

`I also note inconsistencies in `True` vs `true`, use of backticks (which won't work in OAS files when rendered in ReSpec, you have to use <code> tags instead, not referencing specs using HTTP URLs (VCDM v2), and not wrapping type names in code blocks. Please make a pass to fix all of that because it'll create spec debt that we'll just have to clean up later.`

Addressed in commit - https://github.com/w3c-ccg/vc-api/pull/510/commits/f4b15badd83dba5e5a7e642c1a7813168659178c

bparth24 commented:
@bparth24 to review [Verifiable Credential Schema](https://github.com/w3c-ccg/vc-api/blob/main/components/VerifiableCredential.yml)
- ensure it has all the fields required
- Ideally, Verifiable Presentation will use VC as a sub-schema (clean up results object)
- remove summary object

bparth24 commented:
> The group discussed this on the 2025-08-12 telecon.
> 
> @bparth24 to review [Verifiable Credential Schema](https://github.com/w3c-ccg/vc-api/blob/main/components/VerifiableCredential.yml)
> 
> * ensure it has all the fields required
> * Ideally, Verifiable Presentation will use VC as a sub-schema (clean up results object)
> * remove summary object

Addressed all points as discussed above in this commit https://github.com/w3c-ccg/vc-api/pull/510/commits/0f22f53fadea28f58e02d78f1f255d8afae8bdf6.

@msporny cc: @TallTed @dlongley Ready for review.
